MediaBrowser.Api/Playback/Hls/DynamicHlsService.cs

```diff
@@ -962,24 +965,43 @@ protected override string GetVideoArguments(StreamState state, EncodingOptions e
 
                 //args += " -mixed-refs 0 -refs 3 -x264opts b_pyramid=0:weightb=0:weightp=0";
 
+                var hasGraphicalSubs = state.SubtitleStream != null && !state.SubtitleStream.IsTextSubtitleStream && state.SubtitleDeliveryMethod == SubtitleDeliveryMethod.Encode;
+
+                var hasCopyTs = false;
+
                 // Add resolution params, if specified
                 if (!hasGraphicalSubs)
                 {
-                    args += EncodingHelper.GetOutputSizeParam(state, encodingOptions, codec, true);
+                    var outputSizeParam = EncodingHelper.GetOutputSizeParam(state, encodingOptions, codec);
+
+                    args += outputSizeParam;
+
+                    hasCopyTs = outputSizeParam.IndexOf("copyts", StringComparison.OrdinalIgnoreCase) != -1;
                 }
 
-                // This is for internal graphical subs
+                // This is for graphical subs
                 if (hasGraphicalSubs)
                 {
-                    args += EncodingHelper.GetGraphicalSubtitleParam(state, encodingOptions, codec);
+                    var graphicalSubtitleParam = EncodingHelper.GetGraphicalSubtitleParam(state, encodingOptions, codec);
+
+                    args += graphicalSubtitleParam;
+
+                    hasCopyTs = graphicalSubtitleParam.IndexOf("copyts", StringComparison.OrdinalIgnoreCase) != -1;
+                }
+
+                if (!hasCopyTs)
+                {
+                   args += " -copyts";
                 }
 
-                //args += " -flags -global_header";
-            }
+                args += " -avoid_negative_ts disabled";
 
-            if (args.IndexOf("-copyts", StringComparison.OrdinalIgnoreCase) == -1)
-            {
-                args += " -copyts";
+                if (!(state.SubtitleStream != null && state.SubtitleStream.IsExternal && !state.SubtitleStream.IsTextSubtitleStream))
```

Comment:
The subtitles and video will out of sync when burning external graphic subtitles. So don't add `-start_at_zero` only in this condition.

Reply:
I wonder what happens if we drop `-start_at_zero` altogether. We need to go over all those options and review them, some of them aren't probably needed...

Reply:
This might be related to the web client requesting position zero before it requests the actual position it wants.

Reply:
No, it doesn't. Those are unrelated zeros :)

Reply:
After testing, I found that `-start_at_zero` is necessary to use with `-ss` when seeking, otherwise the target position cannot be determined. Except for the above case.

Reply:
Looks like a bug in `ffmpeg` to me TBH. Can we add your explanation above as the comment in the source?

Reply:
Ok.  With the phone at my side, I might not be able to easily add comments now.